source/uwp/Renderer/lib/Util.cpp

```diff
@@ -144,29 +144,45 @@ HRESULT GenerateActionsProjection(
     for (auto& containedAction : containedActions)
     {
         ComPtr<ABI::AdaptiveCards::Uwp::IAdaptiveActionElement> projectedContainedAction;
-        switch (containedAction->GetElementType())
-        {
-            case ActionType::OpenUrl:
-                RETURN_IF_FAILED(MakeAndInitialize<::AdaptiveCards::Uwp::AdaptiveOpenUrlAction>(&projectedContainedAction,
-                    std::static_pointer_cast<AdaptiveCards::OpenUrlAction>(containedAction)));
-                break;
-            case ActionType::ShowCard:
-                RETURN_IF_FAILED(MakeAndInitialize<::AdaptiveCards::Uwp::AdaptiveShowCardAction>(&projectedContainedAction,
-                    std::static_pointer_cast<AdaptiveCards::ShowCardAction>(containedAction)));
-                break;
-            case ActionType::Submit:
-                RETURN_IF_FAILED(MakeAndInitialize<::AdaptiveCards::Uwp::AdaptiveSubmitAction>(&projectedContainedAction,
-                    std::static_pointer_cast<AdaptiveCards::SubmitAction>(containedAction)));
-                break;
-            default:
-                return E_UNEXPECTED;
-                break;
-        }
+        RETURN_IF_FAILED(GenerateActionProjection(containedAction, &projectedContainedAction));
+
         RETURN_IF_FAILED(projectedParentContainer->Append(projectedContainedAction.Detach()));
     }
     return S_OK;
 } CATCH_RETURN;
 
+HRESULT GenerateActionProjection(
+    const std::shared_ptr<AdaptiveCards::BaseActionElement> action,
+    ABI::AdaptiveCards::Uwp::IAdaptiveActionElement** projectedAction) noexcept try
+{
+    if (action == nullptr)
+    {
+        projectedAction = nullptr;
```

Comment:
>projectedAction [](start = 8, length = 15)

I believe what you want to do is *projectedAction = nullptr. That way whatever is inside the ComPtr is null, instead of the smart pointer object itself.

Reply:
Ah good catch. It actually works regardless since projectedAction is nullptr by default, but totally right that this line of code isn't assigning the right pointer :) Fixed and re-tested!